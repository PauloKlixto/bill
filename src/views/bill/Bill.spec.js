/* eslint-disable no-undef */
import { shallowMount, createLocalVue } from '@vue/test-utils'
import mockProducts from '@/views/bill/assets/products'
import Bill from './Bill'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

const store = new Vuex.Store({
  modules: {
    products: {
      namespaced: true,
      actions: {
        fetchProducts: jest.fn(),
        fetchBudget: jest.fn(),
        onPutValue: jest.fn()
      },
      state: {
        products: mockProducts,
        billBudget: 100,
        value: 100
      },
      getters: {}
    }
  }
})

describe('Bill', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Bill, { store, localVue })
  })

  test('Componente existe', () => {
    expect(wrapper.exists()).toBe(true)
  })

  describe('Methods', () => {
    const selectedVideoGame = {
      id: '3',
      model: 'videoGame',
      image: 'videoGame',
      name: 'Video Game',
      value: 60,
      hasQuantityLimit: false,
      maxQuantity: 0
    }

    test('Method - appendValue', () => {
      wrapper.setData({
        form: {
          yacht: 9
        }
      })

      wrapper.vm.appendValue('yacht')
    })

    test('Method - prependValue', () => {
      wrapper.setData({
        form: {
          yacht: 9
        }
      })

      wrapper.vm.prependValue('yacht')
    })

    test('Method - deleteOperation', () => {
      wrapper.setData({
        form: {
          barOfGold: 398404505
        }
      })

      let event = {
        inputType: 'deleteContentBackward'
      }

      wrapper.vm.deleteOperation('barOfGold', event)
      expect(wrapper.vm.form['barOfGold']).toBe(39840450)
    })

    test('Method - operatorWhenIsBiggerThanValue', () => {
      wrapper.setData({
        form: {
          videoGame: 2
        }
      })

      wrapper.vm.operatorWhenIsBiggerThanValue(selectedVideoGame, 'videoGame')
      expect(wrapper.vm.form['videoGame']).toBe(1)
    })

    describe('Method - changeValue', () => {
      let mockOperatorWhenIsBiggerThanValue = jest.fn(),
        mockIsItemWithLimit = jest.fn()

      test('When model is empty', () => {
        wrapper.setData({
          form: {
            videoGame: ''
          }
        })

        wrapper.vm.changeValue('videoGame')
        expect(wrapper.vm.form['videoGame']).toBe(0)
      })

      test('When operator is bigger than value', () => {
        wrapper.setMethods({
          operatorWhenIsBiggerThanValue: mockOperatorWhenIsBiggerThanValue
        })

        wrapper.setData({
          form: {
            videoGame: 2
          }
        })

        wrapper.vm.changeValue('videoGame')
        expect(mockOperatorWhenIsBiggerThanValue).toHaveBeenCalledWith(
          selectedVideoGame,
          'videoGame'
        )
      })

      test('When operator is NOT bigger than value', () => {
        wrapper.setMethods({
          isItemWithLimit: mockIsItemWithLimit
        })

        wrapper.setData({
          form: {
            videoGame: 1
          }
        })

        wrapper.vm.changeValue('videoGame')
        expect(mockIsItemWithLimit).toHaveBeenCalledWith(
          selectedVideoGame,
          'videoGame'
        )
      })
    })

    test('Method - putValue', () => {
      wrapper.vm.putValue()
    })
  })

  test('Componente nÃ£o alterado', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
