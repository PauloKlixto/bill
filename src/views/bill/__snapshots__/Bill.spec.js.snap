// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bill Componente n√£o alterado 1`] = `
"<section class=\\"Bill\\">
  <section class=\\"Bill__billGatesArea\\"><img src=\\"@/assets/images/billGates.png\\" class=\\"Bill__billPhoto\\">
    <h1>Spend Bill Gates' Money</h1>
  </section>
  <basebudget-stub value=\\"$ 100\\"></basebudget-stub>
  <section class=\\"Bill__cardsWrapper\\">
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Big Mac</h2> <span class=\\"Bill__cardsValue\\">$ 2</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Cup of Coffee</h2> <span class=\\"Bill__cardsValue\\">$ 4</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Book</h2> <span class=\\"Bill__cardsValue\\">$ 15</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Video Game</h2> <span class=\\"Bill__cardsValue\\">$ 60</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Charity</h2> <span class=\\"Bill__cardsValue\\">$ 100</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Headphones</h2> <span class=\\"Bill__cardsValue\\">$ 200</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Air Jordans</h2> <span class=\\"Bill__cardsValue\\">$ 200</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Skateboard</h2> <span class=\\"Bill__cardsValue\\">$ 300</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Smartphone</h2> <span class=\\"Bill__cardsValue\\">$ 600</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Gaming Console</h2> <span class=\\"Bill__cardsValue\\">$ 600</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Bike</h2> <span class=\\"Bill__cardsValue\\">$ 800</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Drone</h2> <span class=\\"Bill__cardsValue\\">$ 800</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Designer Handbag</h2> <span class=\\"Bill__cardsValue\\">$ 1,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Jet Ski</h2> <span class=\\"Bill__cardsValue\\">$ 8,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">4K TV</h2> <span class=\\"Bill__cardsValue\\">$ 10,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Diamond Ring</h2> <span class=\\"Bill__cardsValue\\">$ 10,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Rolex</h2> <span class=\\"Bill__cardsValue\\">$ 15,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Speed Boat</h2> <span class=\\"Bill__cardsValue\\">$ 30,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Food Truck</h2> <span class=\\"Bill__cardsValue\\">$ 50,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Tesla</h2> <span class=\\"Bill__cardsValue\\">$ 70,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Monster Truck</h2> <span class=\\"Bill__cardsValue\\">$ 150,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Helicopter</h2> <span class=\\"Bill__cardsValue\\">$ 175,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Ferrari</h2> <span class=\\"Bill__cardsValue\\">$ 200,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Lamborghini</h2> <span class=\\"Bill__cardsValue\\">$ 200,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Firetruck</h2> <span class=\\"Bill__cardsValue\\">$ 200,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Townhouse</h2> <span class=\\"Bill__cardsValue\\">$ 200,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Bar</h2> <span class=\\"Bill__cardsValue\\">$ 300,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Pizza Shop</h2> <span class=\\"Bill__cardsValue\\">$ 500,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Bar of Gold</h2> <span class=\\"Bill__cardsValue\\">$ 500,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Superbowl Ad</h2> <span class=\\"Bill__cardsValue\\">$ 5,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Beach House</h2> <span class=\\"Bill__cardsValue\\">$ 5,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Yacht</h2> <span class=\\"Bill__cardsValue\\">$ 10,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">F16</h2> <span class=\\"Bill__cardsValue\\">$ 15,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Skyscraper</h2> <span class=\\"Bill__cardsValue\\">$ 50,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Mansion</h2> <span class=\\"Bill__cardsValue\\">$ 50,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Rocket</h2> <span class=\\"Bill__cardsValue\\">$ 60,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Passenger Jet</h2> <span class=\\"Bill__cardsValue\\">$ 150,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Monalisa</h2> <span class=\\"Bill__cardsValue\\">$ 780,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">Cruise Ship</h2> <span class=\\"Bill__cardsValue\\">$ 1,200,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">NBA Team</h2> <span class=\\"Bill__cardsValue\\">$ 1,300,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">MBL Team</h2> <span class=\\"Bill__cardsValue\\">$ 1,500,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
    <div class=\\"Bill__cards\\"><img src=\\"\\" class=\\"Bill__cardsImage\\">
      <h2 class=\\"Bill__cardsTitle\\">NFL Team</h2> <span class=\\"Bill__cardsValue\\">$ 2,300,000,000</span>
      <basebutton-stub tag=\\"button\\" label=\\"Sell\\" color=\\"rgb(245, 59, 87)\\" disabled=\\"true\\" class=\\"Bill__cardsSell\\"></basebutton-stub> <input class=\\"Bill__cardsInput\\">
      <basebutton-stub tag=\\"button\\" label=\\"Buy\\" color=\\"rgb(5, 196, 107\\" class=\\"Bill__cardsBuy\\"></basebutton-stub>
    </div>
  </section>
  <section class=\\"Bill__resume\\" style=\\"display: none;\\">
    <h2>Your Shopping Spree</h2>
    <ul class=\\"Bill__resumeList\\"></ul>
  </section>
</section>"
`;
